app/assets/javascripts/social_networking/controllers/profile-controller.js

```diff
@@ -2,48 +2,48 @@
   "use strict";
 
   // Provide interaction with a participant's profile.
-  function ProfileCtrl(profileId, Profiles, Nudges) {
+  function ProfileCtrl(alertService, profileId, Profiles, Nudges) {
       var self = this;
-      self._profiles = Profiles;
-      self._nudges = Nudges;
-      self.profile = {};
-      self._profiles.getOne(profileId)
-      .then(function(profile) {
-          self.id = profile.id;
-          self.profile = profile;
-          self.iconSrc = '';
+      this._profiles = Profiles;
+      this._nudges = Nudges;
+      this.alertService = alertService;
```

Comment:
Refactor suggestion so that the view logic doesn't know about implementation:

```javascript
this.getAlerts = function() {
  return alertService.alerts;
}
```